tests/Unit/Services/OnboardingServiceTest.php

```
@@ -0,0 +1,311 @@
+        $dailyStats = $analytics['daily_stats'];
+        $this->assertGreaterThan(0, $dailyStats->count());
+
+        // 各日付グループ内に複数のイベントタイプがあることを確認
+        $firstDayStats = $dailyStats->first();
+        $this->assertGreaterThan(0, $firstDayStats->count());
+    }
+
+    #[\PHPUnit\Framework\Attributes\Test]
+    public function it_updates_user_progress_safely_with_transaction(): void
+    {
+        $initialProgress = $this->user->onboarding_progress;
```

PHPMD (2.15.0)

Avoid unused local variables such as '$initialProgress'. (Unused Code Rules)

(UnusedLocalVariable)